dashboard/app/controllers/admin_search_controller.rb

```diff
@@ -13,21 +13,38 @@ def lookup_section
 
   def search_for_teachers
     SeamlessDatabasePool.use_persistent_read_connection do
-      email_filter = "%#{params[:emailFilter]}%"
-      address_filter = "%#{params[:addressFilter]}%"
+      @teachers = User.where(user_type: 'teacher')
 
-      # TODO(asher): Determine whether we should be doing an inner join or a left
-      # outer join.
-      @teachers = User.where(user_type: 'teacher').where("email LIKE ?", email_filter).where("full_address LIKE ?", address_filter).joins(:followers).group('followers.user_id')
-
-      # If requested, join with the workshop_attendance table to filter out based
-      # on PD attendance.
+      # If requested, filter...
+      if params[:emailFilter].present?
+        @teachers = @teachers.where("email LIKE ?", "%#{params[:emailFilter]}%")
+      end
+      if params[:addressFilter].present?
+        @teachers = @teachers.where("full_address LIKE ?", "%#{params[:addressFilter]}%")
+      end
+      # TODO(asher): Improve PD filtering.
       if params[:pd] == "pd"
         @teachers = @teachers.joins("INNER JOIN workshop_attendance ON users.id = workshop_attendance.teacher_id").distinct
       elsif params[:pd] == "nopd"
         @teachers = @teachers.joins("LEFT OUTER JOIN workshop_attendance ON users.id = workshop_attendance.teacher_id").where("workshop_attendance.teacher_id IS NULL").distinct
       end
 
+      # If requested, filter away users that have unsubscribed.
+      if params[:unsubscribe].present?
+        unsubscribers = [].tap do |results|
```

Comment:
If this is going to be used often, maybe we should have a job cache the unsubscriber list in the dashboard db periodically?